scene/main/node.cpp

```diff
@@ -925,10 +925,9 @@ bool Node::_validate_node_name(String &p_name) {
 void Node::set_name(const String &p_name) {
 
 	String name = p_name;
-	_validate_node_name(name);
-
+        ERR_FAIL_COND(!_validate_node_name(name));
```

Comment:
That's not the correct fix, `_validate_node_name` is meant to fix the name given as input, not error out.

Reply:
Well that's up for discussion, but the current logic is that if you pass an invalid node name, this method will fix it for you and use the fixed name.

Reply:
The method name is really misleading ...